Composer/packages/client/src/utils/authClient.ts

```diff
@@ -123,14 +125,20 @@ async function getARMTokenForTenant(tenantId: string): Promise<string> {
     const { __csrf__ = '' } = window;
     options.headers['X-CSRF-Token'] = __csrf__;
   }
-
-  const result = await fetch(`/api/auth/getARMTokenForTenant?tenantId=${tenantId}`, options);
-  if (result.status >= 400) {
-    const data = await result.json();
-    throw Error(data.error?.diagnostics?.description || 'get ARM token failure');
+  // do we have a valid token in the cache for this tenant?
+  if (getTokenFromCache(`token-${tenantId}`)) {
+    return getTokenFromCache(`token-${tenantId}`);
   } else {
-    const { accessToken = '' } = await result.json();
-    return accessToken;
+    const result = await fetch(`/api/auth/getARMTokenForTenant?tenantId=${tenantId}`, options);
+    if (result.status >= 400) {
+      const data = await result.json();
+      throw Error(data.error?.diagnostics?.description || 'get ARM token failure');
```

Comment:
Nit: most of the time we throw new Error, but in other places (less often) we have throw Error

Reply:
good catch throw new Error is the right way i think!